src/com/amazon/ion/impl/lite/IonStructLite.java

```diff
@@ -629,38 +584,17 @@ public void put(String fieldName, IonValue value)
     }
 
     @Override
-    public ListIterator<IonValue> listIterator(int index) {
-        return new SequenceContentIterator(index, isReadOnly()) {
-            @Override
-            public void remove() {
-                if (__readOnly) {
-                    throw new UnsupportedOperationException();
-                }
-                force_position_sync();
-
-                int idx = __pos;
-                if (!__lastMoveWasPrevious) {
-                    // position is 1 ahead of the array index
-                    idx--;
-                }
-                if (idx < 0) {
-                    throw new ArrayIndexOutOfBoundsException();
-                }
-
-                IonValueLite concrete = __current;
-                int concrete_idx = concrete._elementid();
-                assert(concrete_idx == idx);
-
-                if (_field_map != null) {
-                    remove_field_from_field_map(concrete.getFieldName(), idx);
-                }
-                super.remove();
+    void beforeIteratorRemove(IonValueLite value, int idx) {
+        if (_field_map != null) {
```

Comment:
(Question) why not have this check inline instead of adding separate layer/method to check for this since its only used in struct?

Reply:
If I'm reading it right I think that's what he meant by "Removing this override resulted in a speedup of up to 13% depending on the dataset."

This would make a good comment though, explaining the motivation for this callback pattern.

Reply:
The reason is mostly to avoid having to expose private IonStructLite internals to IonContainerLite (an ancestor class). `_field_map` and the methods called from `beforeIteratorRemove` and `afterIteratorRemove` are `private` and should remain that way. I suppose we could have these methods be unique to IonStructLite rather than overriding from IonContainerLite, but then we have to `instanceof` and cast at the call site. That doesn't feel as clean.